src/index.js

```diff
@@ -1,14 +1,58 @@
 #!/usr/bin/env node
-
+/* eslint-disable no-console */
 const core = require('@actions/core');
 const exec = require('@actions/exec');
+const fs = require('fs');
+const path = require('path');
+const fetch = require('node-fetch');
+
+async function getGithubOidcToken(audience = 'api://AzureADTokenExchange') {
+  const requestToken = process.env.ACTIONS_ID_TOKEN_REQUEST_TOKEN;
+  const requestUrl = process.env.ACTIONS_ID_TOKEN_REQUEST_URL;
+
+  if (!requestToken || !requestUrl) {
+    throw new Error('OIDC environment variables are missing. Ensure id-token: write permission is enabled.');
+  }
+
+  const res = await fetch(`${requestUrl}&audience=${encodeURIComponent(audience)}`, {
+    headers: {
+      Authorization: `Bearer ${requestToken}`,
+    },
+  });
+
+  if (!res.ok) {
+    throw new Error(`Failed to get OIDC token: ${res.status} ${await res.text()}`);
+  }
+
+  const { value } = await res.json();
+  return value;
+}
 
 async function run() {
   try {
     const configDir = core.getInput('configDir');
     const mode = core.getInput('mode');
     const subscriptionId = core.getInput('subscriptionId');
+    const clientId = core.getInput('clientId');
+    const tenantId = core.getInput('tenantId');
+    const clientSecret = core.getInput('clientSecret');
 
+    process.env.AZURE_CLIENT_ID = clientId;
+    process.env.AZURE_TENANT_ID = tenantId;
+    process.env.AZURE_SUBSCRIPTION_ID = subscriptionId;
+    process.env.AZURE_CLIENT_SECRET = clientSecret;
+    // Detect auth scheme: OIDC (federated) or Service Principal
+    if (process.env.GITHUB_ACTIONS && process.env.ACTIONS_ID_TOKEN_REQUEST_URL) {
```

Comment:
As per ticket, if we also had an input of `authScheme` we could be a little more aggressive about checking provided inputs are a compatible set, rather than guessing based on what's provided.  Just a thought.